Zhongli.Bot/Modules/RoleModule.cs

```diff
@@ -454,6 +460,51 @@ public enum FilterType
             [HelpSummary("Include users whose nickname matches this regex pattern, and if they don't have one, their username. Ignores case.")]
             public string? NameRegexPattern { get; set; }
 
+            [HelpSummary("Include users who joined the server after the specified datetime.")]
+            public string? JoinedAfter { get; set; }
+
+            [HelpSummary("Include users who joined the server before the specified datetime.")]
+            public string? JoinedBefore { get; set; }
+
+            [HelpSummary("Include users who joined the server at the specified datetime.")]
+            public string? JoinedAt { get; set; }
+
+            [HelpSummary("Include users who joined the server at or after the specified datetime.")]
+            public string? JoinedAtAfter { get; set; }
+
+            [HelpSummary("Include users who joined the server at or before the specified datetime.")]
+            public string? JoinedAtBefore { get; set; }
+
+            [HelpSummary("Include users whose Discord account was created after the specified datetime.")]
+            public string? CreatedAfter { get; set; }
+
+            [HelpSummary("Include users whose Discord account was created before the specified datetime.")]
+            public string? CreatedBefore { get; set; }
+
+            [HelpSummary("Include users whose Discord account was created at the specified datetime.")]
+            public string? CreatedAt { get; set; }
+
+            [HelpSummary("Include users whose Discord account was created at or after the specified datetime.")]
+            public string? CreatedAtAfter { get; set; }
+
+            [HelpSummary("Include users whose Discord account was created at or before the specified datetime.")]
+            public string? CreatedAtBefore { get; set; }
+
+            [HelpSummary("Include users who boosted the server after the specified datetime.")]
+            public string? BoosterAfter { get; set; }
+
+            [HelpSummary("Include users who boosted the server before the specified datetime.")]
+            public string? BoosterBefore { get; set; }
+
+            [HelpSummary("Include users who boosted the server at the specified datetime.")]
+            public string? BoosterAt { get; set; }
+
+            [HelpSummary("Include users who boosted the server at or after the specified datetime.")]
+            public string? BoosterAtAfter { get; set; }
+
+            [HelpSummary("Include users who boosted the server at or before the specified datetime.")]
+            public string? BoosterAtBefore { get; set; }
```

Comment:
Use `DateTimeOffset?`, there is a TypeReader for it included by default, so you wouldn't have to parse it yourself, along with validation.

See included Typreaders in here https://docs.stillu.cc/guides/commands/typereaders.html

Reply:
Updated. Thank for the review!